tests/driver_lis2dh12/main.c

```diff
@@ -11,170 +11,699 @@
  * @{
  *
  * @file
- * @brief       Test application for the LIS2DH12 accelerometer driver
+ * @brief       Test application for LIS2DH12 accelerometer driver
  *
- * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ * @author      Jan Mohr <jan.mohr@ml-pa.com>
  *
  * @}
  */
 
-#include <assert.h>
-#include <stdio.h>
-
-#include "fmt.h"
+#include "stdio.h"
+#include "string.h"
+#include "stdlib.h"
 #include "xtimer.h"
-#include "mutex.h"
+#include "fmt.h"
+#include "thread.h"
+#include "shell.h"
+
 #include "lis2dh12.h"
 #include "lis2dh12_params.h"
+#include "lis2dh12_registers.h"
 
- /* delay between sensor data reads */
-#define DELAY       (100UL * US_PER_MS)
+#define ENABLE_DEBUG 0
+#include "debug.h"
 
-/* allocate some memory to hold one formatted string for each sensor output, so
- * one string for the X, Y, and Z reading, respectively */
-static char str_out[3][8];
+#define REFERENCE_DEFAULT 10 /* LSB according to SCALE */
+
+#define THOLD_SHOCK_MILLIG_DEFAULT 1500
+#define NUM_DATA_SHOCK_DETECT 7  /* detect shock in NUM last FIFO samples */
```

Comment:
Where does that number come from?

Reply:
it is only for testing, to not check all values, only the 7 newest.